api_tests/lib/omnilayer.js

```diff
@@ -0,0 +1,232 @@
+    messagePrefix: "\x18Bitcoin Signed Message:\n",
+    bech32: "bcrt",
+    bip32: {
+        public: 0x043587cf,
+        private: 0x04358394,
+    },
+    pubKeyHash: 0x6f,
+    scriptHash: 0xc4,
+    wif: 0xef,
+};
+
+let _rpc_client;
+
+function createOmniRpcClient() {
+    const config = global.harness.ledgers_config.omnilayer;
+    if (!config) {
+        throw new Error("ledger.omnilayer configuration is needed");
+    }
+    return (_rpc_client =
+        _rpc_client ||
+        new BitcoinRpcClient({
+            network: "regtest",
+            port: config.rpc_port,
+            host: config.rpc_host,
+            username: config.rpc_username,
+            password: config.rpc_password,
+        }));
+}
+
+module.exports.getBalance = async function(tokenId, address) {
+    const res = await _rpc_client.command([
+        {
+            method: "omni_getbalance",
+            parameters: [address, tokenId],
+        },
+    ]);
+    return res[0].balance;
+};
+
+module.exports.createClient = () => {
+    return createOmniRpcClient();
+};
+
+module.exports.generate = async function(num = 1) {
+    return createOmniRpcClient().generate(num);
+};
+
+module.exports.activateSegwit = async function() {
+    return createOmniRpcClient().generate(432);
+};
+
+module.exports.swaperoo = async function(aliceDetails, bobDetails, tokenId, omni_value, btc_value) {
+    if (!tokenId) {
+        throw new Error("tokenId must be provided, got: " + tokenId);
+    }
+    // Caveat: If alice has a lot of BTC on her omni output it will all go to Bob
+
+    // alice_output = prev output for omni = new output for BTC (should only have dust BTC)
+    const { alice_keypair, alice_omni_utxo, alice_final_address } = aliceDetails;
+    // bob_btc_output = prev output for BTC = output for BTC change
+    // bob_omni_output = new output for Omni
+    const { bob_keypair, bob_btc_utxo, bob_btc_output, bob_final_address } = bobDetails;
+
+    const alice_output = bitcoin.address.toOutputScript(alice_final_address, regtest);
+    const bob_omni_output = bitcoin.address.toOutputScript(bob_final_address, regtest);
+
+    if (bob_omni_output === bob_btc_output) {
+        throw new Error("Bob BTC and Omni output MUST be different");
+    }
+
+    const txb = new bitcoin.TransactionBuilder();
+
+    const fee = 2500;
+    const dust = 500;
+    const bob_btc = bob_btc_utxo.value - fee / 2 - btc_value - dust;
+    const alice_btc = alice_omni_utxo.value - fee / 2 + btc_value;
+
+    // Alice Omni input
+    txb.addInput(alice_omni_utxo.txid, alice_omni_utxo.vout, null, alice_output);
+    // Bob BTC input
+    txb.addInput(bob_btc_utxo.txid, bob_btc_utxo.vout, null, bob_btc_output);
+
+    // Add BTC change back to Bob
+    txb.addOutput(bob_btc_output, bob_btc);
+    // Add BTC output to Alice
+    txb.addOutput(alice_output, alice_btc);
+    // Add Omni output to Bob (it's Omni because it's different from the inputs)
+    txb.addOutput(bob_omni_output, dust);
```

Comment:
❓this can be/should be `0` right?
Your test run with 0 at least.

Reply:
Ah yes indeed, we do not need dust for the `OP_RETURN` output. I got mistaken. Thanks

Reply:
Hum, I expect a minimum dust amount to be present on this UTXO or the transaction would be rejected.
Indeed, it did not get rejected when I changed it. This is unexpected, I'll do more research.

Reply:
Ok, the issue is that dust checks are deactivated on regtest by default. Now sorted, see https://github.com/comit-network/comit-rs/issues/691#issuecomment-462998340 for details